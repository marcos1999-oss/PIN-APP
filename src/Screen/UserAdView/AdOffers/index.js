import React from "react";
import { View, Text, Button } from "native-base";
import styles, { purpleText } from "./style";
import CustomIcon from "../../../CustomIcon";
import AdPerformance from "../../../Components/AdPerformance";
import AdMenu from "../../../Components/AdMenu";

export default class AdOffers extends React.Component {
  render() {
    const iconName = {
      'regular': 'radar',
      'holiday': 'celebrate',
      'birthday': 'birthday',
    }[this.props.post.kinds];

    const iconColor = {
      'regular': 'black',
      'holiday': '#02BAFF',
      'birthday': '#AD378C',
    }[this.props.post.kinds];

    return (
      <View style={styles.container}>
        <View style={styles.eventContainer}>
          { iconName && (
            <Button rounded style={styles.eventButton}>
              <CustomIcon name={ iconName } style={{ ...styles.eventIcon, color: iconColor }} />
            </Button>
          )}

          <Text style={{ ...styles.eventText, color: iconColor, textTransform: 'capitalize' }}>{ this.props.post.kinds }</Text>
        </View>

        <AdPerformance {...this.props} />
        <AdMenu {...this.props} />
      </View>
    );
  }
}
