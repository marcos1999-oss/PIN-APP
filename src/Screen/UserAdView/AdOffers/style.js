import { StyleSheet } from "react-native";
import {
  width,
  verticalScale,
  moderateScale,
  scale,
} from "../../../Utils/scaling";

const blackText = "#6E6969";
const green = "#3FC94F";
export const purpleText = "#A338FF";

const styles = StyleSheet.create({
  container: {
    flexDirection: "column",
  },
  eventContainer: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: verticalScale(15),
  },
  eventButton: {
    width: verticalScale(30),
    height: verticalScale(30),
    marginRight: scale(10),
    backgroundColor: "white",
    shadowColor: "#000",
    shadowOffset: { width: 0, height: 3 },
    shadowOpacity: 0.16,
    shadowRadius: 6,
    elevation: 1,
    justifyContent: "center",
  },
  eventIcon: {
    color: "#AD378C",
    fontSize: moderateScale(13),
  },
  eventText: {
    color: "#AD378C",
    fontSize: moderateScale(10),
    fontWeight: "bold",
  },
  adPerformance: {
    flexDirection: "row",
    alignItems: "center",
    marginBottom: verticalScale(23),
  },
  adStatusText: {
    color: blackText,
    fontSize: moderateScale(12),
    fontWeight: "bold",
  },
  performanceContainer: {
    flexDirection: "row",
    marginLeft: scale(22),
    alignItems: "center",
  },
  likeIcon: {
    fontSize: moderateScale(6),
    color: "#DF0103",
  },
  performanceText: {
    marginLeft: scale(5),
    fontSize: moderateScale(10),
    color: blackText,
  },
  watchIcon: {
    fontSize: moderateScale(6),
    color: "#000",
  },
  menuContainer: {
    flexDirection: "column",
  },
  menuItem: {
    flexDirection: "row",
    alignItems: "flex-start",
    justifyContent: "space-between",
    marginBottom: verticalScale(16),
  },
  menuName: {
    fontSize: moderateScale(14),
    color: blackText,
    flex: 6,
  },
  rightText: {
    flex: 2,
    flexDirection: "row",
    alignItems: "flex-start",
    justifyContent: "flex-end",
    alignSelf: "center",
  },
  time: {
    fontSize: moderateScale(10),
    fontWeight: "bold",
    color: green,
  },
  cost: {
    fontSize: moderateScale(14),
    fontWeight: "bold",
    color: blackText,
  },
  percentageIcon: {
    color: green,
    fontSize: moderateScale(14),
  },
  off: {
    fontSize: moderateScale(8),
    fontWeight: "bold",
    color: green,
  },
  horizontalDivider: {
    position: "relative",
    left: scale(-20),
    width,
    height: 0,
    borderBottomWidth: 0.5,
    borderColor: "#707070",
    marginBottom: verticalScale(8),
  },
  vipTitle: {
    fontSize: moderateScale(8),
    fontWeight: "bold",
    color: purpleText,
    marginBottom: verticalScale(7),
  },
});

export default styles;
