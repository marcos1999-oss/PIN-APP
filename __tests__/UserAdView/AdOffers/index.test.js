/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import AdOffers from '../../../src/Screen/UserAdView/AdOffers';

jest.mock('react-native-cached-images', () => {
  return {
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

test('renders correctly', () => {
  const post = {
    kind: {},
  };
  const tree = renderer.create(<AdOffers post={post} />).toJSON();
  expect(tree).toMatchSnapshot();
});
