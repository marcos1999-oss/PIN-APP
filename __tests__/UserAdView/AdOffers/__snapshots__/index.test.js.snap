// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Object {
        "flexDirection": "column",
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 30,
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": undefined,
            "fontSize": 15,
            "fontWeight": "bold",
            "textTransform": "capitalize",
          },
        ]
      }
      uppercase={false}
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 46,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "color": "#6E6969",
            "fontSize": 18,
            "fontWeight": "bold",
          },
        ]
      }
      uppercase={false}
    >
      Live Offers
    </Text>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": 44,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#5F92F3",
              "fontSize": 9,
            },
            Object {
              "fontFamily": "PinAppIcon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        юдк
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#6E6969",
              "fontSize": 15,
              "marginLeft": 10,
            },
          ]
        }
        uppercase={false}
      >
        0
         People Liked This
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginLeft": 44,
          },
        ]
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Object {
              "color": "#000",
              "fontSize": 9,
            },
            Object {
              "fontFamily": "PinAppIcon",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        юдй
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "#6E6969",
              "fontSize": 15,
              "marginLeft": 10,
            },
          ]
        }
        uppercase={false}
      >
        0
         Views
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "flexDirection": "column",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {},
          Object {
            "height": 80,
          },
        ]
      }
    />
  </View>
</View>
`;
